docs/package_templates/cmake_package/all/conanfile.py

```diff
@@ -70,11 +69,12 @@ def configure(self):
         self.settings.rm_safe("compiler.libcxx")
 
     def layout(self):
-        # src_folder must use the same source folder name the project
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        # prefer self.requires method instead of requires attribute
+        # Prefer self.requires method instead of requires attribute
+        # Set transitive_headers=True (which usually also requires transitive_libs=True) if
+        # the dependency is used in any of the packaged header files.
         self.requires("dependency/0.8.1")
```

Comment:
Maybe add an example of a version range fake dependency, which a comment linking to the FAQ for the allowed ones?